src/main/app/claims/claimModelConverter.ts

```diff
@@ -1,63 +1,148 @@
 import DraftClaim from 'drafts/models/draftClaim'
 import { PartyDetails } from 'forms/models/partyDetails'
+import { PartyType } from 'forms/models/partyType'
+import { IndividualDetails } from 'forms/models/individualDetails'
+import { CompanyDetails } from 'forms/models/companyDetails'
+import { OrganisationDetails } from 'forms/models/organisationDetails'
+import { SoleTraderDetails } from 'forms/models/soleTraderDetails'
+import ClaimData from 'claims/models/claimData'
+import { Party } from 'claims/models/details/yours/party'
+import { Individual as ClaimantAsIndividual } from 'claims/models/details/yours/individual'
+import { Company as ClaimantAsCompany } from 'claims/models/details/yours/company'
+import { SoleTrader as ClaimantAsSoleTrader } from 'claims/models/details/yours/soleTrader'
+import { TheirDetails } from 'claims/models/details/theirs/theirDetails'
+import { Organisation as ClaimantAsOrganisation } from 'claims/models/details/yours/organisation'
+import { Individual as DefendantAsIndividual } from 'claims/models/details/theirs/individual'
+import { Company as DefendantAsCompany } from 'claims/models/details/theirs/company'
+import { SoleTrader as DefendantAsSoleTrader } from 'claims/models/details/theirs/soleTrader'
+import { Organisation as DefendantAsOrganisation } from 'claims/models/details/theirs/organisation'
+import InterestDate from 'app/claims/models/interestDate'
+import { Address } from 'claims/models/address'
+import { Address as AddressForm } from 'forms/models/address'
 
 export class ClaimModelConverter {
 
-  static convert (draftClaim: DraftClaim): any {
-    // Remove optional field if empty
-    if (draftClaim.interestDate.date.asString() === '') {
-      delete draftClaim.interestDate.date
-    } else {
-      draftClaim.interestDate.date = draftClaim.interestDate.date.asString() as any
+  static convert (draftClaim: DraftClaim): ClaimData {
+    let claimData: ClaimData = new ClaimData()
+    claimData.interest = draftClaim.interest
+    claimData.externalId = draftClaim.externalId
+    claimData.interestDate = this.convertInterestDate(draftClaim)
+    claimData.amount = draftClaim.amount
+    claimData.claimant = this.convertClaimantDetails(draftClaim)
+    claimData.defendant = this.convertDefendantDetails(draftClaim)
+    claimData.payment = draftClaim.claimant.payment
+    claimData.reason = draftClaim.reason.reason as any
```

Comment:
why as any?

Reply:
this was how it was in the before refactoring, was not sure why, so kept it, will remove that.

Reply:
it was `as any` because it had the type of reason, now it has the type of string so its not needed